ksql-common/src/main/java/io/confluent/ksql/schema/ksql/types/SqlTypes.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql.types;
+
+import io.confluent.ksql.schema.ksql.SqlBaseType;
+
+public final class SqlTypes {
```

Comment:
super nit: i feel like this could be merged with `SqlType` so that I can just have `SqlType.BOOLEAN` (similar to `Schema.OPTIONAL_STRING_SCHEMA`)

Reply:
Nar, I don't like abstract types or interfaces then also then being factory classes.  They shouldn't know about their subclasses / impls

Reply:
Humm.... maybe I don't mind either way...

Reply:
OK, so I took a look at doing this and even the compiler throws a warning:

> Referencing subclass blah from superclass SqlType initializer might lead to classloader deadlock.

That's enough for me to err to the way it currently is. Either that or I change it to an interface, rather than an abstract class.  (I'd prefer an interface anyway)